presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ExchangeNode.java

```diff
@@ -91,11 +97,14 @@ public ExchangeNode(
 
         checkArgument(scope != REMOTE || type == Type.REPARTITION || !partitioningScheme.isReplicateNullsAndAny(), "Only REPARTITION can replicate remotely");
 
+        checkArgument(type == Type.GATHER || !orderingScheme.isPresent(), "ordering scheme can be set only for GATHERING");
```

Comment:
Do you know that `PartitioningScheme` allows for `orderingScheme` to be present when `type=GATHER`?